plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -186,20 +186,67 @@ instance PrettyDefaultBy config Integer => PrettyBy config ExBudget where
         , "}"
         ]
 
-data ExBudgetState exBudgetCat = ExBudgetState
-    { _exBudgetStateTally  :: ExTally exBudgetCat -- ^ for counting what cost how much
-    , _exBudgetStateBudget :: ExBudget -- ^ for making sure we don't spend too much
-    }
+-- | @(-)@ on 'ExCPU'.
+minusExCPU :: ExCPU -> ExCPU -> ExCPU
+minusExCPU = coerce $ (-) @Integer
+
+-- | @(-)@ on 'ExMemory'.
+minusExMemory :: ExMemory -> ExMemory -> ExMemory
+minusExMemory = coerce $ (-) @Integer
+
+-- | Subtract an 'ExBudget' from an 'ExRestrictingBudget'.
+minusExBudget :: ExRestrictingBudget -> ExBudget -> ExRestrictingBudget
+ExRestrictingBudget (ExBudget cpuL memL) `minusExBudget` ExBudget cpuR memR =
+    ExRestrictingBudget $ ExBudget (cpuL `minusExCPU` cpuR) (memL `minusExMemory` memR)
+
+-- | A budgeting mode to execute an evaluator in.
+data ExBudgetMode
+    = Counting                         -- ^ For calculating the cost of execution.
+    | Tallying                         -- ^ For a detailed report on what costs how much +
+                                       --     the same overall budget that 'Counting' gives.
+    | Restricting ExRestrictingBudget  -- ^ For execution, to avoid overruns.
+
```

Comment:
Just to be really niggly, the constructors are in a different order for `ExBudgetMode` and `ExBudgetState`.

Reply:
I can't keep two data types standing next to each other in sync and I was supposed to keep two CEK machines in sync when updating costing / builtin applications / etc stuff! Thanks, fixed.